website/app/styles/pages/application/sidebar.scss

```diff
@@ -115,3 +93,64 @@ body.application:not(.index) {
     }
   }
 }
+
+
+// ------------------------------------------
+
+
+// ANIMATION (for small viewports)
+
+// HOME PAGE
+
+body.application.index {
+
+  // by default we remove the sidebar from the render tree, so its children can't receive focus
+  .doc-page-sidebar {
+    display: none;
+    transition: none;
+  }
+
+  // when the `isSidebarInRenderTree` class is added to the `<body>` (via JS) we add the sidebar to the render tree (so we can then animate it)
+  &.isSidebarInRenderTree .doc-page-sidebar {
+    display: block;
+    transform: translate3d(0, -100%, 0);
+    transition: transform 0.25s ease-in;
+  }
+
+  &.isSidebarVisibleOnSmallViewport .doc-page-sidebar {
+    transform: translate3d(0, 0, 0);
+    transition: transform 0.25s ease-out;
+  }
+}
+
+// ALL THE OTHER PAGES
+
+body.application:not(.index) {
+
+  // the sidebar is animated only on small viewports
+  @include breakpoint.with-mobile-sidebar() {
+
+    // by default we remove the sidebar from the render tree, so its children can't receive focus
+    .doc-page-sidebar {
+      display: none;
+      transition: none;
+    }
+
+    // when the `isSidebarInRenderTree` class is added to the `<body>` (via JS) we add the sidebar to the render tree (so we can then animate it)
+    &.isSidebarInRenderTree .doc-page-sidebar {
+      display: flex;
+      transform: translate3d(-100%, 0, 0);
+      transition: transform 0.25s ease-in;
+    }
+
+    &.isSidebarVisibleOnSmallViewport .doc-page-sidebar {
+      transform: translate3d(0, 0, 0);
+      transition: transform 0.25s ease-out;
```

Comment:
Nit: I would try to avoid repeating these styles maybe by using a sass variable or a mixin or something.